src/main/java/com/iridium/iridiumskyblock/managers/SchematicManager.java

```diff
@@ -115,4 +125,78 @@ private CompletableFuture<Void> pasteSchematic(Island island, Schematics.Schemat
         return completableFuture;
     }
 
+    public boolean processTransaction(Player player, Schematics.SchematicConfig schematic) {
+
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+        Optional<Island> island = IridiumSkyblock.getInstance().getTeamManager().getTeamViaID(user.getTeamID());
+
+        if(!island.isPresent()){
+            //NO ISLAND
+            return false;
+        }
+
+        TeamBank bank;
+
+        double bankBalance;
+        List<Placeholder> bankPlaceholders;
+        double moneyCost = schematic.regenCost.money;
+        Economy economy = IridiumSkyblock.getInstance().getEconomy();
+
+        if(moneyCost == 0 || economy != null && economy.getBalance(player) >= moneyCost) {
+            //INVALID TRANSACTION
+            return false;
+        }
+
+        for (String schematicBankItem : schematic.regenCost.bankItems.keySet()) {
+
+            bank = IridiumSkyblock.getInstance().getTeamManager().getTeamBank(island.get(), schematicBankItem);
+            bankBalance = bank.getNumber();
+
+            double cost = round(schematic.regenCost.bankItems.get(schematicBankItem), 2);
+
+            if (bank.getNumber() < cost) {
+                player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotAfford
+                        .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+                IridiumSkyblock.getInstance().getSchematics().failSound.play(player);
+
+                return false;
+            }
```

Comment:
This needs to be moved outside the for loop, so we check if they can afford it before we start taking stuff away